src/detext/run_detext.py

```diff
@@ -2,275 +2,162 @@
 """
 
-import argparse
 import logging
 import os
 import sys
 import time
+from typing import NamedTuple, List
 
 import tensorflow as tf
 import tensorflow_ranking as tfr
+from arg_suite.arg_suite import arg_suite
 from detext.train import train
 from detext.train.data_fn import input_fn
 from detext.utils import misc_utils, logger, executor_utils
 
 
-def add_arguments(parser):
-    """Build ArgumentParser."""
-    parser.register("type", "bool", lambda v: v.lower() == "true")
-
+@arg_suite
+class Args(NamedTuple):
+    """
+    DeText: a Deep Text understanding framework for NLP related ranking, classification, and language generation tasks.
+    It leverages semantic matching using deep neural networks to
+    understand member intents in search and recommender systems.
+    As a general NLP framework, currently DeText can be applied to many tasks,
+    including search & recommendation ranking, multi-class classification and query understanding tasks.
+    """
+    #  kwargs utility funcation for split a str to a List, provided for backward compatibilities
+    #  Please use built-in List parsing support when adding new arguments
+    __comma_split_list = lambda t: {'type': lambda s: [t(item) for item in s.split(',')] if ',' in s else t(s), 'nargs': None}  # noqa: E731
     # network
-    parser.add_argument("--ftr_ext", choices=['cnn', 'bert', 'lstm_lm', 'lstm'], help="NLP feature extraction module.")
-    parser.add_argument("--num_units", type=int, default=128, help="word embedding size.")
-    parser.add_argument("--num_units_for_id_ftr", type=int, default=128, help="id feature embedding size.")
-    parser.add_argument("--sp_emb_size", type=int, default=1, help="Embedding size of sparse features")
-    parser.add_argument("--num_hidden", type=str, default='0', help="hidden size.")
-    parser.add_argument("--num_wide", type=int, default=0, help="number of wide features per doc.")
-    parser.add_argument("--num_wide_sp", type=int, default=None, help="number of sparse wide features per doc")
-    parser.add_argument("--use_deep", type=str2bool, default=True, help="Whether to use deep features.")
-    parser.add_argument("--elem_rescale", type=str2bool, default=True,
-                        help="Whether to perform elementwise rescaling.")
-    parser.add_argument("--use_doc_projection", type=str2bool, default=False,
-                        help="whether to project multiple doc features to 1 vector.")
-    parser.add_argument("--use_usr_projection", type=str2bool, default=False,
-                        help="whether to project multiple usr features to 1 vector.")
+    ftr_ext: str  # NLP feature extraction module.
+    _ftr_ext = {'choices': ['cnn', 'bert', 'lstm_lm', 'lstm']}
+    num_units: int = 128  # word embedding size.
+    num_units_for_id_ftr: int = 128  # id feature embedding size.
+    sp_emb_size: int = 1  # Embedding size of sparse features
+    num_hidden: List[int] = 0  # hidden size.
```

Comment:
we need to convert the input as the List of int, same for other input parameters such as the filter_window_sizes, emb_sim_func, filter_window_sizes, all_metrics, score_rescale

Reply:
The current test cases do not cover this so they are not captured by pytest or related sh files.

Reply:
can you talk in more details? i'm not sure what you mean here

Reply:
In this implementation, there is a function `__comma_split_list` to split comma separated inputs and convert them to a list. Some of them are converted, such as `feature_names` but a few are left out. I listed some of the remaining ones here in the comments.

Reply:
I see. Thanks for catching this!
We'll have same issues for the boolean variables as well since the type conversion only happens for `_use_deep`.

Reply:
`_use_deep` is covered in the PR. Do we still use `use_wide` because I saw it in the test case of `li-detext`?

Reply:
I meant that for other boolean params such as elem_rescale, use_horovod, etc., the lower case tolerance is not added.

For use_wide, as I remember, this param was removed long time ago. If users don't want to use it, they just don't include "wide_ftrs" in the feature_names.